features/fixtures/mazerunner/app/src/main/java/com/bugsnag/android/mazerunner/MainActivity.kt

```diff
@@ -73,6 +76,33 @@ class MainActivity : Activity() {
         log("MainActivity.onResume complete")
     }
 
+    private fun setMazeRunnerAddress() {
+        val context = MazerunnerApp.applicationContext()
+        val externalFilesDir = context.getExternalFilesDir(null)
+        val configFile = File(externalFilesDir, "fixture_config.json")
+        log("Attempting to read Maze Runner address from config file ${configFile.path}")
+
+        // Poll for the fixture config file
+        val pollStart = System.currentTimeMillis()
+        while (System.currentTimeMillis() - pollStart < CONFIG_FILE_TIMEOUT) {
```

Comment:
nit: assigning an "end time" avoids the arithmetic in the polling check

```suggestion
        val pollEnd = System.currentTimeMillis() + CONFIG_FILE_TIMEOUT
        while (System.currentTimeMillis() < pollEnd) {
```